File: OnlineRepairBooking/frontend/src/api/order.js
import request from '@/utils/request'

export const createOrder = (data) => {
  return request({
    url: '/orders',
    method: 'post',
    data
  })
}

export const getOrderList = (params) => {
  return request({
    url: '/orders',
    method: 'get',
    params
  })
}

export const getOrderDetail = (id) => {
  return request({
    url: `/orders/${id}`,
    method: 'get'
  })
}

export const cancelOrder = (id, cancelReason) => {
  return request({
    url: `/orders/${id}/cancel`,
    method: 'post',
    data: { cancelReason }
  })
}

export const grabOrder = (id) => {
  return request({
    url: `/worker/orders/${id}/grab`,
    method: 'post'
  })
}

export const acceptOrder = (id) => {
  return request({
    url: `/worker/orders/${id}/accept`,
    method: 'post'
  })
}

export const startService = (id) => {
  return request({
    url: `/worker/orders/${id}/start`,
    method: 'post'
  })
}

export const completeService = (id) => {
  return request({
    url: `/worker/orders/${id}/complete`,
    method: 'post'
  })
}

export const getPendingOrders = (params) => {
  return request({
    url: '/worker/orders/pending',
    method: 'get',
    params
  })
}

export const getWorkerOrders = (params) => {
  return request({
    url: '/worker/orders',
    method: 'get',
    params
  })
}

File: OnlineRepairBooking/backend/internal/routes/order.go
package routes

import (
	"online-repair-booking/internal/handlers"
	"online-repair-booking/internal/middleware"
	"online-repair-booking/pkg/database"

	"github.com/labstack/echo/v4"
)

func RegisterOrderRoutes(g *echo.Group) {
	db := database.MySQL
	orderHandler := handlers.NewOrderHandler(db)
	bidHandler := handlers.NewBidHandler(db)

	userGroup := g.Group("/orders", middleware.JWTAuth())
	{
		userGroup.POST("", orderHandler.CreateOrder)
		userGroup.GET("", orderHandler.GetOrderList)
		userGroup.GET("/:id", orderHandler.GetOrderDetail)
		userGroup.POST("/:id/cancel", orderHandler.CancelOrder)
		userGroup.GET("/:id/bids", bidHandler.GetOrderBids)
		userGroup.POST("/bids/accept", bidHandler.AcceptBid)
	}

	workerGroup := g.Group("/worker/orders", middleware.JWTAuth())
	{
		workerGroup.GET("", orderHandler.WorkerGetOrderList)
		workerGroup.GET("/pending", orderHandler.GetPendingOrders)
		workerGroup.POST("/:id/accept", orderHandler.WorkerAcceptOrder)
		workerGroup.POST("/:id/start", orderHandler.WorkerStartService)
		workerGroup.POST("/:id/complete", orderHandler.WorkerCompleteService)
		workerGroup.POST("/:id/grab", bidHandler.PlaceBid)
	}

	adminGroup := g.Group("/admin/orders", middleware.JWTAuth())
	{
		adminGroup.POST("/auto-dispatch", bidHandler.AutoDispatch)
	}
}
